### PR TITLE
Spectral Averaging Incorporation

### PR DESCRIPTION
**Pull request by nbollis**
Added spectral averaging Extensions project to handle the averaging of MsDataScans and MzSpectrum. This project also handles the processing methods to determine how to average the spectra

Added equals override method to MzSpectrum Object
98.9% test coverage. Woot. Woot.

### CODE REVIEW

**Comment by trishorts** [comment]
concerned that newtonsoft nuget doesn't seem to support .net core 
https://www.nuget.org/packages/Newtonsoft.Json/

**Comment by avcarr2** [comment]
> concerned that newtonsoft nuget doesn't seem to support .net core https://www.nuget.org/packages/Newtonsoft.Json/

Newtsonsoft does support .NET Standard 2.0 though. [Doesn't .NET Standard cover .NET and .NET Core version 2.0, 2.1, 2.2, 3.0, 3.1, 5.0, and 6.0?](https://docs.microsoft.com/en-us/dotnet/standard/net-standard?tabs=net-standard-2-0)

**Review by avcarr2 — DISMISSED** [review]
Interface to mzLib looks really good. Seems very easy to use and extend in the future.

**Review by avcarr2 — APPROVED** [review]

File: mzLib/MassSpectrometry/MzSpectra/MzSpectrum.cs [hunk 1]
```diff
@@ -787,6 +787,20 @@ public double CalculateDotProductSimilarity(MzSpectrum spectrumToCompare, Tolera
...
+            MzSpectrum typedSpectrum = (MzSpectrum)spectrumToCompare;
+            if (!typedSpectrum.XArray.SequenceEqual(XArray))
+                return false;
+            if (!typedSpectrum.YArray.SequenceEqual(YArray))
+                return false;
+            
+            return true;
+        }
+
         private bool Peak2satisfiesRatio(double peak1theorIntensity, double peak2theorIntensity, double peak1intensity, double peak2intensity, double intensityRatio)
```
[the remark this answers is not in the document]
Comment (nbollis) [reply mzLib/MassSpectrometry/MzSpectra/MzSpectrum.cs:813]: Implemented in recent commit

**Review by avcarr2 — APPROVED** [review]

**Review by Alexander-Sol — APPROVED** [review]

**Review by Alexander-Sol — APPROVED** [review]

**Review by avcarr2 — APPROVED** [review]

**Pull request merged**